### PR TITLE
Add support for hosted connector

### PR DESCRIPTION
**Pull request by mthenw**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -21,22 +21,19 @@ class EventGateway {
...
     }
 
-    if (!config.connectorUrl.startsWith('http')) {
-      config.connectorUrl = 'http://' + config.connectorUrl
-    }
-
     if (isHosted(configuration.url)) {
       config.eventsUrl = configuration.url.startsWith('http') ? configuration.url : `https://${configuration.url}`
       config.configurationUrl = hostedGenerateConfigUrl(config.eventsUrl)
+      config.connectorUrl = config.configurationUrl
```
Comment (alexdebrie) [inline lib/index.js]: We used to check on whether the passed URLs included a protocol. Should we add that here?

Also, for the hosted, config + connector hit the same URL but are routed to different ports at the proxy?

File: lib/index.js [hunk 2]
```diff
@@ -21,22 +21,19 @@ class EventGateway {
...
       config.space = hostedParseSpace(config.eventsUrl)
       config.accessKey = configuration.accessKey
     } else {
       config.eventsUrl = configuration.url.startsWith('http') ? configuration.url : `http://${configuration.url}`
       config.configurationUrl = configuration.configurationUrl || generateConfigUrl(config.eventsUrl)
+      config.connectorUrl = configuration.connectorUrl || generateConnectorUrl(config.eventsUrl)
```
Comment (alexdebrie) [inline lib/index.js]: Same question as above.

File: lib/index.js (hunk 1 above)
Comment (mthenw) [reply lib/index.js]: the url here is generated by the code so it include the protocol

Comment (alexdebrie) [reply lib/index.js]: Good point

**Pull request merged**